ComponentKit/Utilities/CKArrayControllerChangeset.mm

```diff
@@ -348,20 +348,26 @@ static void _iterate(CK::ArrayController::Input::Items::ItemsBucketizedBySection
 }
 
 /**
- Clients of Output::Changeset::map() may reuturn invalid pairs. For example {nil, <object>} for a deletion, instead of
+ Clients of Output::Changeset::map() may return invalid pairs. For example {nil, <object>} for a deletion, instead of
  {<object>, nil}.
  */
-NS_INLINE void _validateBeforeAfterPair(Output::Changeset::BeforeAfterPair pair, IndexPath indexPath, CKArrayControllerChangeType changeType)
+NS_INLINE void _validateBeforeAfterPair(Output::Changeset::BeforeAfterPair pair, IndexPath sourceIndexPath, IndexPath destinationIndexPath, CKArrayControllerChangeType changeType)
 {
   if (changeType == CKArrayControllerChangeTypeUpdate) {
-    CKInternalConsistencyCheckIf(pair.first != nil, ([NSString stringWithFormat:@"update {%zd, %zd}: before MUST NOT be nil.", indexPath.item, indexPath.section]));
-    CKInternalConsistencyCheckIf(pair.second != nil, ([NSString stringWithFormat:@"update {%zd, %zd}: after MUST NOT be nil.", indexPath.item, indexPath.section]));
+    CKInternalConsistencyCheckIf(pair.first != nil, ([NSString stringWithFormat:@"update {%zd, %zd}: before MUST NOT be nil.", sourceIndexPath.item, sourceIndexPath.section]));
+    CKInternalConsistencyCheckIf(pair.second != nil, ([NSString stringWithFormat:@"update {%zd, %zd}: after MUST NOT be nil.", sourceIndexPath.item, sourceIndexPath.section]));
+    CKInternalConsistencyCheckIf(sourceIndexPath.section != NSNotFound, ([NSString stringWithFormat:@"update MUST have sourceIndexPath"]));
```

Comment:
Sadly I cannot compare it to `{}` directly, so options here are `IndexPath()` and `IndexPath(NSNotFound, NSNotFound)`. I prefet the latter, since it makes it obvious it's comparing to invalid indexPath.